From: Benjamin Berg <bberg@redhat.com>
Date: Wed, 22 Jun 2022 17:37:26 +0200
Subject: goodix: Make fingerlist parse error non-fatal

Otherwise we cannot recover from the error by doing a device reset.
---
 libfprint/drivers/goodixmoc/goodix_proto.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/libfprint/drivers/goodixmoc/goodix_proto.c b/libfprint/drivers/goodixmoc/goodix_proto.c
index b615dba..72511a8 100644
--- a/libfprint/drivers/goodixmoc/goodix_proto.c
+++ b/libfprint/drivers/goodixmoc/goodix_proto.c
@@ -393,10 +393,8 @@ gx_proto_parse_body (uint16_t cmd, uint8_t *buffer, uint16_t buffer_len, pgxfp_c
                                        fingerid_length,
                                        &presp->finger_list_resp.finger_list[num]) != 0)
             {
-              g_error ("parse fingerlist error");
-              presp->finger_list_resp.finger_num = 0;
-              presp->result = GX_FAILED;
-              break;
+              g_warning ("Failed to parse finger list");
+              return -1;
             }
           offset += fingerid_length;
         }
